atomicwrites/__init__.py

```diff
@@ -19,12 +19,27 @@ def _path_to_unicode(x):
 
 
 if sys.platform != 'win32':
+    def _sync_directory(directory):
+        # Ensure that filenames are written to disk
+        fd = os.open(directory, 0)
+        try:
+            os.fsync(fd)
+        finally:
+            os.close(fd)
+
     def _replace_atomic(src, dst):
         os.rename(src, dst)
+        _sync_directory(os.path.dirname(dst))
 
     def _move_atomic(src, dst):
         os.link(src, dst)
         os.unlink(src)
```

Comment:
Oh wait, I didn't actually write anything in the README: Basically I consider it somewhat acceptable if the file is linked twice in that timewindow between link and unlink. People mostly care about `dst` "appearing" atomically, IMO.